### PR TITLE
Sign in

### PR DESCRIPTION
**Pull request by diversifiedmechanic**
I probably over-committed. I had a lot of back and forth with how I should handle everything. 
On the front-end: alert letting the User know the last time they logged in and sends them back to 'home'. If it's an incorrect username / password, it sends them to the 'incorrect user' page. If they get a 404 from the server, there is an alert letting them know to try again in a little while. 
Back-end: Server is checking for right username/password, if not found -> sends a 401.
Something happens during the query -> sends a 404.
correct username / password, pulls the last log-in, username, userId -> sends to user. Updates last login to the current time.

### CODE REVIEW

File: db/controllers/user.js [hunk 1]
```diff
@@ -1,29 +1,40 @@
...
+                SET
+                  last_login = TO_TIMESTAMP($1)
+                WHERE
+                  id = $2
+              `, [(Date.now() / 1000), results.rows[0].id])
+              .then(() => {
+                client.release();
+                cb(null, results.rows);
+              });
+          }
```
Comment (diversifiedmechanic) [inline db/controllers/user.js:37]: I'm not a fan of this. Pretty clunky but it works.

File: server/routes/profile/index.js [hunk 2]
```diff
@@ -52,4 +52,18 @@ router.delete('/:id', (req, res) => {
...
+  db.attemptSignIn(email, password, (err, result) => { // will be sent to hashing algo first
+    if (err) {
+      res.status(404).send(err.detail);
+    } else if (result === 'invalid') {
+      res.sendStatus(401);
+    } else {
+      res.status(200).json(result);
+    }
+  });
+});
```
Comment (diversifiedmechanic) [inline server/routes/profile/index.js:67]: I should probably be throwing a new Error? Can adjust if that would be a better solution.

File: src/ApiServices/attemptSignInApi.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import axios from 'axios';
+
+function attemptSignInApi(email, password) {
+  return axios.post('/api/profile/signin', { email, password })
+    .then((response) => response.data[0]);
+}
+
+export default attemptSignInApi;
```
Comment (naszzz) [inline src/ApiServices/attemptSignInApi.js:8]: wow. think i have been unnecessarily literally writing all my API services funcs to 'return new Promise.. etc' ... this is so much better.

File: server/routes/profile/index.js (hunk 2 above)
Comment (naszzz) [reply server/routes/profile/index.js:67]: I think the error handling is fine.

File: src/components/SignIn.jsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+      }) // a message that they have logged in? or just change the icon?
+      .catch((error) => {
+        if (error.response.status === 401) {
+          toPage('incorrectUser');
+        } else {
+          alert('Oh no! Something went wrong, please wait then try again.');
+        }
+      });
+  };
+
```
Comment (naszzz) [inline src/components/SignIn.jsx:29]: lgtm

**Pull request merged**